garak/generators/openai.py

```diff
@@ -82,58 +82,62 @@
 }
 
 
-class OpenAIGenerator(Generator):
-    """Generator wrapper for OpenAI text2text models. Expects API key in the OPENAI_API_KEY environment variable"""
+class OpenAICompatible(Generator):
+    """Generator base class for OpenAI compatible text2text restful API. Implements shared initialization and execution methods."""
+
+    ENV_VAR = "OpenAICompatible_API_KEY".upper()  # Placeholder override when extending
 
     supports_multiple_generations = True
-    generator_family_name = "OpenAI"
+    generator_family_name = "OpenAICompatible"  # Placeholder override when extending
 
+    # template defaults optionally override when extending
     temperature = 0.7
     top_p = 1.0
     frequency_penalty = 0.0
     presence_penalty = 0.0
     stop = ["#", ";"]
 
+    def _load_client(self):
+        # Required stub implemented when extending `OpenAICompatible`
+        raise NotImplementedError
+
+    def _clear_client(self):
+        # Required stub implemented when extending `OpenAICompatible`
+        raise NotImplementedError
+
     def __init__(self, name, generations=10):
         self.name = name
-        self.fullname = f"OpenAI {self.name}"
+        self.fullname = f"{self.generator_family_name} {self.name}"
 
         super().__init__(name, generations=generations)
 
-        api_key = os.getenv("OPENAI_API_KEY", default=None)
-        if api_key is None:
+        self.api_key = os.getenv(self.ENV_VAR, default=None)
+        if self.api_key is None:
             raise ValueError(
-                'Put the OpenAI API key in the OPENAI_API_KEY environment variable (this was empty)\n \
-                e.g.: export OPENAI_API_KEY="sk-123Xgenerators/openai.pyXXXXXXXXXXX"'
+                f'Put the {self.generator_family_name} API key in the {self.ENV_VAR} environment variable (this was empty)\n \
+                e.g.: export {self.ENV_VAR}="sk-123Xgenerators/openai.pyXXXXXXXXXXX"'
```

Comment:
weird old paste typo
```suggestion
                e.g.: export {self.ENV_VAR}="sk-123XXXXXXXXXXXX"'
```